sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3429,7 +3491,7 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
    * Replaces [[UnresolvedDeserializer]] with the deserialization expression that has been resolved
    * to the given input attributes.
    */
-  object ResolveDeserializer extends Rule[LogicalPlan] {
+  class ResolveDeserializer(catalog: SessionCatalog) extends Rule[LogicalPlan] {
```

Comment:
this `catalog` doesn't appear to get used; can we remove it?

Reply:
Done